Content.Shared/Mobs/Systems/MobStateSystem.cs

```diff
@@ -20,15 +20,26 @@ public partial class MobStateSystem : EntitySystem
     private ISawmill _sawmill = default!;
 
     private EntityQuery<MobStateComponent> _mobStateQuery;
+    private EntityQueryEnumerator<MobStateComponent> _mobStateQueryEnumerator;
 
     public override void Initialize()
     {
         _sawmill = _logManager.GetSawmill("MobState");
         _mobStateQuery = GetEntityQuery<MobStateComponent>();
         base.Initialize();
         SubscribeEvents();
+
+        SubscribeLocalEvent<MobStateComponent, MapInitEvent>(OnMapInit);
     }
 
+    private void OnMapInit(EntityUid mob, MobStateComponent comp, MapInitEvent args)
+    {
+        // tells other components that the mobstate is what it be now it do
+        var curstate = comp.CurrentState;
+        var ev = new MobStateChangedEvent(mob, comp, curstate, curstate);
+        RaiseLocalEvent(mob, ev, true);
+
+    }
```

Comment:
Raising dummy events on it is a massive performance sink, just subscribe to compinit on the other components.

Reply:
Done!
~or at least done for the application I needed. There's like 20 subscriptions to MobStateChangedEvent and I didn't vet any of the others since they mostly seem like they don't need it. 🤷